qutebrowser/mainwindow/tabbedbrowser.py

```diff
@@ -526,18 +526,21 @@ def _get_new_tab_idx(self, related):
         else:
             pos = config.val.tabs.new_position.unrelated
         if pos == 'prev':
-            idx = self._tab_insert_idx_left
-            # On first sight, we'd think we have to decrement
-            # self._tab_insert_idx_left here, as we want the next tab to be
-            # *before* the one we just opened. However, since we opened a tab
-            # *before* the currently focused tab, indices will shift by
-            # 1 automatically.
+            if config.val.tabs.new_position.stacking:
+                idx = self._tab_insert_idx_left
+                # On first sight, we'd think we have to decrement
+                # self._tab_insert_idx_left here, as we want the next tab to be
+                # *before* the one we just opened. However, since we opened a tab
+                # *before* the currently focused tab, indices will shift by
+                # 1 automatically.
+            else:
+                idx = self.widget.currentIndex()
         elif pos == 'next':
             if config.val.tabs.new_position.stacking:
                 idx = self._tab_insert_idx_right
+                self._tab_insert_idx_right += 1
             else:
                 idx = self.widget.currentIndex() + 1
-            self._tab_insert_idx_right += 1
```

Comment:
I'm not trying to jerk you round, honest! But I just noticed something.

* If you have just one tab, say wikipedia, and new_position.related set to next and stacking set to true, then open a couple of related tabs you will have [wiki] [wiki1] [wiki2]
* Then if you set stacking to false an open another one you will have [wiki] [wiki3] [wiki1] [wiki2]
* Then set stacking to true again and open another one and you will have [wiki] [wiki3] [wiki1] [wiki4] [wiki2]

So now things look a bit jumbled. Constrast that to when related is set to prev:
* [wiki2] [wiki1] [wiki]
* [wiki2] [wiki1] [wik3] [wiki]
* [wiki4] [wiki2] [wiki1] [wik3] [wiki]

The new related stack opens at the end of the other related stacked ones again! If you move that increment back out of the conditional block it'll work for next to. That's just a suggestion though, when testing something you tend to do the strangest things, because obviously that whole function isn't designed to deal with `pos` changing between calls (like if you set it to `first` both of them should be incremented).

Reply:
Ah, I haven't thought of this corner case, I'll move this out of the if to fix that :)